core/ledger/ledger.go

```diff
@@ -89,21 +89,30 @@ func (l *Ledger) GetBlockWithHeight(height uint32) (*Block, error) {
 func (l *Ledger) GetBlockWithHash(hash Uint256) (*Block, error) {
 	bk, err := l.Store.GetBlock(hash)
 	if err != nil {
-		return nil,NewDetailErr(err, ErrNoCode, "[Ledger],GetBlockWithHeight failed with hash="+ hash.ToString())
+		return nil, NewDetailErr(err, ErrNoCode, "[Ledger],GetBlockWithHeight failed with hash="+hash.ToString())
 	}
 	return bk, nil
 }
 
+//ContainBlock checks if the block existed in ledger
+func (l *Ledger) ContainBlock(hash Uint256) bool {
```

Comment:
No need lock at here, since no write operation for the ledger and the read lock is provided by leveldb in store module.